source/extensions/filters/http/cache/cache_headers_parser.h

```diff
@@ -0,0 +1,102 @@
+#pragma once
+
+#include "envoy/common/time.h"
+#include "envoy/http/header_map.h"
+
+#include "absl/strings/string_view.h"
+#include "absl/time/time.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace HttpFilters {
+namespace Cache {
+
+using OptionalDuration = absl::optional<SystemTime::duration>;
+
+struct RequestCacheControl {
+  bool must_validate = false;
+  bool no_store = false;
+  bool no_transform = false;
+  bool only_if_cached = false;
+  OptionalDuration max_age;
+  OptionalDuration min_fresh;
+  OptionalDuration max_stale;
+
+  bool operator==(const RequestCacheControl& rhs) const {
+    return (must_validate == rhs.must_validate) && (no_store == rhs.no_store) &&
+           (no_transform == rhs.no_transform) && (only_if_cached == rhs.only_if_cached) &&
+           (max_age == rhs.max_age) && (min_fresh == rhs.min_fresh) && (max_stale == rhs.max_stale);
+  }
+};
+
+// std::ostream& operator<<(std::ostream& os, const RequestCacheControl& request_cache_control);
+
+// 'public' directive is used to cache responses that are normally not cache-able
+// or cache requests/responses with authorization headers,
+// since both are not supported now, the 'public' header is ignored
+struct ResponseCacheControl {
+  // must_validate is true if any of 'must-revalidate', 'proxy-revalidate' or 'no-cache' directives
+  // is present. 'no-cache' arguments are ignored for now so it is equivalent to 'must-revalidate'
+  bool must_validate = false;
+
+  // no_store is true if any of 'no-store' or 'private' directives is present.
+  // 'private' arguments are ignored for now so it is equivalent to 'no-store'
+  bool no_store = false;
+
+  bool no_transform = false;
+
+  // max_age is set if to 's-maxage' if present, if not it is set to 'max-age' if present.
+  OptionalDuration max_age;
+
+  bool operator==(const ResponseCacheControl& rhs) const {
+    return (must_validate == rhs.must_validate) && (no_store == rhs.no_store) &&
+           (no_transform == rhs.no_transform) && (max_age == rhs.max_age);
+  }
+};
+
+// std::ostream& operator<<(std::ostream& os, const ResponseCacheControl& response_cache_control);
+
+class CacheHeadersParser {
```

Comment:
The name `CacheHeadersParser` makes me think you're intended to instantiate a parser object and use it to do things, rather than it being a collection of helper functions. Envoy naming convention seems to be to name a collection of static functions a "utility", for example, https://github.com/envoyproxy/envoy/blob/master/source/common/http/path_utility.h

Reply:
Makes sense, will modify it. Do you think I should merge this class with the existing `CacheFilterUtils` class, or keep them separate?